lib/posix-process/arch/arm64/execve.c

```diff
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/arch/ctx.h>
+#include <uk/essentials.h>
+
+void execve_arch_execenv_init(struct ukarch_execenv *execenv,
+			      __uptr ip, __uptr sp)
+{
+	execenv->regs.lr = ip;
+	execenv->regs.sp = sp;
+
```

Comment:
I think we should also set `spsr_el1` as I believe it to have an unsafe value coming from a `malloc`d stack. I would reuse the one of the parent and, additionally ` &= ~PSR_I` so that IRQs are enblaed. Ditto for x86, let's reuse the parents flags but with IRQs forcefully enabled (although I think in both cases it should be safe to assume they are already enabled IMO).

Reply:
Actually, is there a reason for updating `SPSR_EL1` and `ESR_EL1` in `ukarch_execenv_load`?

Reply:
Following an offline discussion, we decided that we should for now keep these semantics for both `ESR_EL1` and `SPSR_EL1`.

So, initially, the ARM code shared the entries for both IRQ's and regular traps. Since the former would not require `ESR_EL1` and potentially creating confusion we would normally restore `ESR_EL1` to make sure that IRQ code does not somehow think it is in a trap because `ESR_EL1` has a value that a trap would normally set. But I will soon submit a small patch that separates the two entries, thus restoring `ESR_EL1` is no longer required.

On the other hand, we still have to keep `SPSR_EL1` updated as that restores the `PSTATE`. For example, consider a page fault that may happen during a syscall. This is essentially a nested trap so `SPSR_EL1` must remain consistent when returning from the page fault as that could mean confusing the previous trap. Same thing if we nest page faults, which we should soon have a use case for.

So, for now, it is not clear if it is really worth dropping `ESR_EL1`, as we cannot see the use of restoring it anymore but, at the same time, it ensures a little bit of consistency on return from exceptions, although we are not sure if it is a useful consistency to have either, it will probably save us in the future. From what I have seen, neither linux not freebsd restore it on exception return but, I do not have knowledge of how they deal with nested pagefaults, or if they do consider them at all.